external-api/api-tests/v1/email.api-test.ts

```diff
@@ -1,49 +1,95 @@
+      version: 24,
+    },
+    uri: 'https://api.notifications.service.gov.uk/v2/notifications/efd12345-1234-5678-9012-ee123456789f',
   },
 };
 
-jest.mock('notifications-node-client', () => {
-  class MockNotifyClient {
-    sendEmail: () => Promise<{ status: number; data: { content: { body: object }; id: string; template: object } }>;
-
-    constructor() {
-      this.sendEmail = async () => Promise.resolve(mockResponse);
-    }
-  }
+// Mock Axios
+const axiosMock = new MockAdapter(axios);
 
-  return {
-    NotifyClient: MockNotifyClient,
-  };
-});
+const mockBody = {
+  templateId: 'tmp1234-1234-5678-9012-abcd12345678',
+  sendToEmailAddress: 'test@testing.com',
+  emailVariables: {
+    name: 'Testing',
+  },
+};
 
 describe('/email', () => {
   describe('POST /v1/email', () => {
-    it('should return response from notify api', async () => {
-      const mockBody = {
-        templateId: '1234',
-        sendToEmailAddress: 'test@testing.com',
-        emailVariables: {
-          name: 'Testing',
-        },
-      };
+    it('should return 201 response from MDM Gov Notify API', async () => {
+      axiosMock.onPost(`${APIM_MDM_URL}emails`).reply(HttpStatusCode.Created, mockSuccessfulResponse.data);
+
+      const { status, body } = await post(mockBody).to('/email');
+
+      expect(status).toEqual(mockSuccessfulResponse.status);
+      expect(body).toEqual(mockSuccessfulResponse.data);
+    });
+
+    it('should return error status from MDM Gov Notify API if error message is missing', async () => {
+      const unknown500Status = Math.floor(Math.random() * 50 + 500);
+      axiosMock.onPost(`${APIM_MDM_URL}emails`).reply(unknown500Status, '');
+
+      const { status, body } = await post(mockBody).to('/email');
+
+      expect(status).toEqual(unknown500Status);
+      expect(body).toEqual({});
+    });
+
+    it('should return error status from MDM Gov Notify API if error message provided', async () => {
+      const errorMessage = 'an error message';
+      const unknown500Status = Math.floor(Math.random() * 50) + 500;
+      axiosMock.onPost(`${APIM_MDM_URL}emails`).reply(unknown500Status, errorMessage);
 
       const { status, body } = await post(mockBody).to('/email');
 
-      expect(status).toEqual(mockResponse.status);
-      expect(body).toEqual(mockResponse.data);
+      expect(status).toEqual(unknown500Status);
+      expect(body).toEqual({});
+    });
+  });
+
+  describe('sendEmail', () => {
+    it('should return 201 response from MDM Gov Notify API', async () => {
+      axiosMock.onPost(`${EXTERNAL_API_URL}/email`).reply(HttpStatusCode.Created, mockSuccessfulResponse.data);
+
+      const response = await sendEmail(mockBody.templateId, mockBody.sendToEmailAddress, mockBody.emailVariables);
+
+      expect(response).toEqual(mockSuccessfulResponse.data);
+    });
+
+    it('should null in case of error', async () => {
```

Comment:
Can we add an assertion for `429` please.

Reply:
Helper sendEmail is very simple, it doesn't check status code and returns null on all errors.